### ISSUE TITLE
Unable to open Python console with `NVDA+control+Z`

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#17391]
Issue found while investigating #17390.

### Steps to reproduce:

* Open a PowerPoint presentation
* Focus a text area (or create it), e.g. on the title
* Press `F2` to edit it
* Press `control+k` to insert a link
* While in the edit field of the dialog that opens, press `NVDA+control+Z` to open Python consol

### Actual behavior:
Python console does not open.
### Expected behavior:
Python console should open.
### NVDA logs, crash dumps and other attachments:
```
ERROR - scriptHandler.executeScript (15:13:36.894) - MainThread (2636):
error executing script: <bound method GlobalCommands.script_activatePythonConsole of <globalCommands.GlobalCommands object at 0x06654C30>> with gesture 'contrôle+NVDA+z'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 300, in executeScript
  File "gui\blockAction.pyc", line 79, in funcWrapper
  File "globalCommands.pyc", line 3405, in script_activatePythonConsole
  File "pythonConsole.pyc", line 259, in updateNamespaceSnapshotVars
  File "api.pyc", line 530, in getCaretPosition
  File "NVDAObjects\window\edit.pyc", line 1107, in makeTextInfo
  File "NVDAObjects\window\edit.pyc", line 895, in __init__
AttributeError: 'NoneType' object has no attribute 'duplicate'

```
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2024.4rc1
#### Windows version:
Windows 10
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
